frame/vesting/src/lib.rs

```diff
@@ -302,121 +407,322 @@ pub mod pallet {
 		///     - Reads: Vesting Storage, Balances Locks, Target Account, Source Account
 		///     - Writes: Vesting Storage, Balances Locks, Target Account, Source Account
 		/// # </weight>
-		#[pallet::weight(T::WeightInfo::force_vested_transfer(MaxLocksOf::<T>::get()))]
+		#[pallet::weight(
+			T::WeightInfo::first_force_vested_transfer(MaxLocksOf::<T>::get())
+			.max(T::WeightInfo::last_force_vested_transfer(MaxLocksOf::<T>::get()))
+		)]
 		pub fn force_vested_transfer(
 			origin: OriginFor<T>,
 			source: <T::Lookup as StaticLookup>::Source,
 			target: <T::Lookup as StaticLookup>::Source,
 			schedule: VestingInfo<BalanceOf<T>, T::BlockNumber>,
 		) -> DispatchResult {
 			ensure_root(origin)?;
-			ensure!(schedule.locked >= T::MinVestedTransfer::get(), Error::<T>::AmountLow);
+			Self::do_vested_transfer(source, target, schedule)
+		}
 
-			let target = T::Lookup::lookup(target)?;
-			let source = T::Lookup::lookup(source)?;
-			ensure!(!Vesting::<T>::contains_key(&target), Error::<T>::ExistingVestingSchedule);
+		/// Merge two vesting schedules together, creating a new vesting schedule that unlocks over
+		/// highest possible start and end blocks. If both schedules have already started the current
+		/// block will be used as the schedule start; with the caveat that if one schedule is finished
+		/// by the current block, the other will be treated as the new merged schedule, unmodified.
+		///
+		/// NOTE: If `schedule1_index == schedule2_index` this is a no-op.
+		/// NOTE: This will unlock all schedules through the current block prior to merging.
+		/// NOTE: If both schedules have ended by the current block, no new schedule will be created
+		/// and both will be removed.
+		///
+		/// Merged schedule attributes:
```

Comment:
this seems like implementation detail leaking into public doc?

Reply:
Perhaps, but I am not sure how to best explain to users what the new schedule will look like. I think it is important users understand because it has the potential to significantly delay when they get there funds